builtInFunctions/logsAndEvents.go

```diff
@@ -17,11 +18,18 @@ func addESDTEntryInVMOutput(vmOutput *vmcommon.VMOutput, identifier []byte, toke
 }
 
 func newEntryForESDT(identifier, tokenID []byte, nonce uint64, value *big.Int, args ...[]byte) *vmcommon.LogEntry {
-	nonceBig := big.NewInt(0).SetUint64(nonce)
+	tokenIdentifier := tokenID
+	if nonce != 0 {
+		nonceBig := big.NewInt(0).SetUint64(nonce)
+		hexEncodedNonce := hex.EncodeToString(nonceBig.Bytes())
+
+		tokenIdentifier = append(tokenIdentifier, []byte("-")...)
+		tokenIdentifier = append(tokenIdentifier, []byte(hexEncodedNonce)...)
+	}
 
 	logEntry := &vmcommon.LogEntry{
 		Identifier: identifier,
-		Topics:     [][]byte{tokenID, nonceBig.Bytes(), value.Bytes()},
+		Topics:     [][]byte{tokenIdentifier, value.Bytes()},
```

Comment:
i don't really like this approach. the actual code was better.

Reply:
Ok I will let as it is